packages/guides-restructured-text/tests/unit/Parser/Productions/InlineMarkup/TextRoleRuleTest.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace phpDocumentor\Guides\RestructuredText\Parser\Productions\InlineMarkup;
+
+use phpDocumentor\Guides\Nodes\InlineToken\EmphasisToken;
+use phpDocumentor\Guides\Nodes\InlineToken\GenericTextRoleToken;
+use phpDocumentor\Guides\Nodes\InlineToken\ValueToken;
+use phpDocumentor\Guides\RestructuredText\TextRoles\DefaultTextRoleFactory;
+
+class TextRoleRuleTest extends StartEndRegexRoleRuleTest
+{
+    private TextRoleRule $rule;
+
+    protected function setUp(): void
+    {
+        $this->rule = new TextRoleRule(new DefaultTextRoleFactory());
+    }
+
+    public function getRule(): StartEndRegexRoleRule
+    {
+        return $this->rule;
+    }
+
+    /**
+     * @return array<int, array<int, array<int, string> | bool>>
+     */
+    public function ruleAppliesProvider(): array
+    {
+        return [
+            [
+                [':role:`something'],
+                true,
+            ],
+            [
+                [':php:class:`something'],
```

Comment:
This is a special role. I think sphinx calls this domains: https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html

The current span parser allows this kind of roles in domains. We are using the `CrossReferenceNode` for this. Not sure how this should work.

Reply:
As we give the oppurtunity to invent any kind of role you can also invent one with a domain. I would not solve this on parser level...